sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -346,3 +403,46 @@ rom_error_t boot_data_read(lifecycle_state_t lc_state, boot_data_t *boot_data) {
       return error;
   }
 }
+
+rom_error_t boot_data_write(const boot_data_t *boot_data) {
+  boot_data_t new_entry = *boot_data;
```

Comment:
Given that `boot_data` is a complex type, this looks like an implicit copy operation. Since `boot_data_entry_write` copies the data into a buffer, I wonder if we should create an `union` type (ignoring endianness concerns).

Reply:
Yes, this is a copy in order not to modify the buffer that is passed in, which probably can be relaxed if we want to. `boot_data_entry_write` also copies the struct to a buffer because `flash_write` expects a `uint32_t` array. I defined `boot_data_buffer_t` as a `struct` as opposed to a `union` because I can't seem to find a conclusive answer whether type punning using unions is safe or unspecified behavior.

E.g., the second example [here](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html#Type-punning) definitely looks like something that can be easily missed in a code review and similar to what we would do after passing the buffer in the union to `flash_write()`. Since I expect writes to be not as performance-critical as reads, I tried to be more conservative here.

Reply:
Sounds good. I think this approach is ok given that the size of the struct is considered small. We may have to reconsider the approach if the structure gets to a size in which we need to look at stack utilization.

Reply:
Sounds good, resolving here, captured in #9056 as a cleanup task.